### PR TITLE
WIP:Added Parenthesis to print statements for py2/3 compat

### PR DESCRIPTION
**Pull request by ibigpapa**
Changed 2 lines to add Parenthesis to print statements.   Setup now runs in py3.6 without errors..

### CODE REVIEW

**Comment by jstypka** [comment]
hey Troy, thanks a lot for the pull request! The software was not tested under Python 3 and I'm not sure if those changes would be enough, but they are definitely in a good direction.

The changes would right now break on Python 2, but if you could add `from __future__ import print_function` at the top of the file, that should make the file compatible with both Python 2 and 3. It would be great if you could also squash all those commits into one for clarity and we can merge it.

@kaplun @jacquerie

**Comment by ibigpapa** [comment]
Yeah found that Keras removed graph support in 1.1.2 updated setup file.  Also found an import change in py3.  I'm working to get it to complete a training cycle on some data currently.  More commits to come.

**Comment by ibigpapa** [comment]
I got it mostly working in Py3.  I'm having issues with train_word2vec wanting a vocabulary but not accepting it as a parameter.

File: magpie/utils.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 from __future__ import division
-
-import cPickle as pickle
+try:
+    import cPickle as pickle
+except ImportError:
+    import _pickle as pickle
```
Comment (jstypka) [inline magpie/utils.py]: I think `pickle` uses `_pickle` underneath by default, so only `import pickle` in the except clause would be enough

File: magpie/base/document.py [hunk 2]
```diff
@@ -34,7 +41,13 @@ def __str__(self):
...
     def compute_wordset(self):
         tokens = WordPunctTokenizer().tokenize(self.text)
-        lowercase = map(unicode.lower, tokens)
+        try:
+            lowercase = map(unicode.lower, tokens)
+        except NameError:
+            if sys.version_info.major == 3:
+                lowercase = map(str.lower, tokens)
+            else:
+                raise
```
Comment (jstypka) [inline magpie/base/document.py]: how about `lowercase = [t.lower() for t in tokens]`?

File: magpie/base/document.py [hunk 3]
```diff
@@ -25,7 +26,13 @@ def __init__(self, doc_id, filepath, text=None):
...
 
             with open(filepath, 'r') as f:
-                self.text = f.read().decode('utf-8')
+                try:
+                    self.text = f.read().decode('utf-8')
+                except AttributeError:
+                    if sys.version_info.major == 3:
+                        self.text = f.read()
+                    else:
+                        raise
```
Comment (jstypka) [inline magpie/base/document.py]: Wouldn't that be cleaner?
```
self.text = f.read()

if getattr(self.text, "decode", False):
     self.text = self.text.decode('utf-8')
```

Comment (ibigpapa) [inline magpie/base/document.py]: yep it would i did the quick n dirty :D as i was trying to get this working for a project I'm working on.

**Comment by jstypka** [comment]
Closing, Python 3 is supported since #80

**Pull request closed without merging**